### ISSUE TITLE
After clicking on the button, pressing  Tab key few times other cards are selected but not latest one

### ISSUE THREAD

**Issue by Wajeed-msft** [MicrosoftDocs/msteams-docs#4730]
Copied bug from: https://github.com/microsoft/AdaptiveCards/issues/6778 

**Details from original post:**
### Target Application

Microsoft Teams

### Application Operating System

Windows

### Schema Version

v1.0

### Problem Description

When any feature related button is clicked from any card e.g. Log a case button from view bsc detail card then log a case utterance is triggered followed by its response card is shown. But unable to know where is the control. After pressing  Tab key few times some of the card is selected but not latest one. Only some times it is on recent card or else some of the card it is selected which is on chat screen. It is difficult to know which card it is selected. And for remaining buttons user need to trigger the feature again. Eg. Log a case.
Alt solution: Shift Tab is going the control in reverse order and most of the times the control is selecting the last message(latest message)

### Expected Outcome

when control is coming back from browser screen card focus should be on latest card.

### Actual Outcome

when control is coming back from browser screen, card focus is not on the latest card.

### Card JSON

```JSON
<Adaptive>{
  "type": "AdaptiveCard",
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "version": "1.0",
  "speak": "Of course! I can help you raise a new case or view your existing cases.",
  "body": [
    {
      "type": "TextBlock",
      "text": "Of course! I can help you raise a new case or view your existing cases.",
      "wrap": true
    }
  ],
  "actions": [
    {
      "type": "Action.OpenUrl",
      "title": "Raise a case",
      "url": "sample url"
    },
    {
      "type": "Action.Submit",
      "title": "View cases",
      "data": {
        "msteams": {
          "type": "imBack",
          "value": "View cases"
        }
      }
    },
    
    {
      "type": "Action.Submit",
      "title": "Cancel",
      "data": {
        "msteams": {
          "type": "imBack",
          "value": "Cancel"
        }
      }
    }
  ]
}</Adaptive>
```

### Repro Steps

_No response_---
#### Document Details

⚠ *Do not edit this section.It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3fb4394d-25b1-8752-3996-fe3a2934732d
* Version Independent ID: 6d391381-89d0-117c-27cc-00043bd26a36
* Content: [Cards reference - Teams](https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-reference)
* Content Source: [msteams-platform/task-modules-and-cards/cards/cards-reference.md](https://github.com/MicrosoftDocs/msteams-docs/blob/master/msteams-platform/task-modules-and-cards/cards/cards-reference.md)
* Product: **msteams**
* GitHub Login: @surbhigupta12
* Microsoft Alias: **surbhigupta12**

**Comment by Meghana-MSFT** [comment]
We are looking into this issue. We will update you

**Comment by Meghana-MSFT** [comment]
@prajwalmanju - We tested the scenario as below :

1. Clicking on Raise a case[Lets say 2nd Card]. Loads given URL in browser.
2. Minimize browser and go back to Teams app
3. When we hit Tab, selection is going to Compose Message area
4. When we hit Shift+Tab, selection is going to the card from which Raise a case was tapped[2nd Card].
![image](https://user-images.githubusercontent.com/81138249/144871706-7e1ef88f-3526-494a-b3a4-85899b1fdc49.png)

Is this what you are also experiencing? Could you please confirm on this?

**Comment by prajwalmanju** [comment]
Hello Team,
Please find the below repro steps for this issue.

1) As per the above JSON, if we click on View BSC details, View BSC  response card is shown. 
2) we can observe that control is not inside text editor and we are not able to use  'SHIFT + Tab' key to get focus on latest card, unable to know where is the control. 
3) After pressing Tab key few times some of the card is selected but not latest one. 
4) Only some times it is on recent card or else some of the card it is selected which is on chat screen. It is difficult to know which card it is selected.

**Comment by Meghana-MSFT** [comment]
Thank you for the repro steps, we are checking this. We will update you

**Comment by Meghana-MSFT** [comment]
When ever we press Tab, we see the control is within the compose message area. Pressing Shift+Tab for the first time does not take the control to latest card. But from the second time onwards shift+Tab takes the focus on to latest card. Attaching the video of the same. It would be helpful if you could provide a recording of how you are seeing it.

https://user-images.githubusercontent.com/81138249/146369365-16aeff0d-84c0-445c-b7bf-8727e3bc046f.mp4

**Comment by prajwalmanju** [comment]
Hello Team,

I have attached the recorded video of the issue, please find the below observation details of the issue.

1)  As per the attached video if we click on View cases details, View cases response card is shown.
we can observe that control is not inside text editor and we are not able to use 'SHIFT + Tab' key to get focus on latest card, unable to know where is the control.
2) After pressing "Shift+Tab" key few times some of the card is selected but not latest one.
and also we can observed that after click on "Shift + tab " key some of the card it is selected which is on chat screen. It is difficult to know which card it is selected

Thanks!

https://user-images.githubusercontent.com/93922557/147361151-651568b0-7ac8-40ef-af4c-77a53d328bd7.mp4

.

**Comment by Meghana-MSFT** [comment]
In our case, Cursor is present in the compose message area and hitting Shift+tab is taking us to messages within the chat area only. We will check this internally.

**Comment by prajwalmanju** [comment]
Hello Meghana,

Thanks for the update, I have gone through your video, in your case also cursor is not present in the compose message area on click of view cases button second time, it looks like you selected the composer message area via mouse. Could you please check once using keyboard keys without using mouse.

Thanks!

**Comment by Meghana-MSFT** [comment]
No, Even though I touched mouse by mistake, I have not placed the cursor in compose message area using mouse. Attaching another video of the same for your reference. 

https://user-images.githubusercontent.com/81138249/147551180-e3b0e5e7-a7c4-4ac6-a25e-e24b6907c8a0.mp4

**Comment by prajwalmanju** [comment]
Hello Meghana,

Thanks for the Update, I have gone through your video it looks like on click of "shift + tab" key it was not selecting latest card (latest card delivered time 14:54, on click of "shift + tab" key focus was at previous card ), could you please observe  control was not on latest card on click of "shift + tab" key, the same thing we have mentioned in the case description as well, please let me know if you have any queries.

Thanks!

**Comment by Meghana-MSFT** [comment]
As mentioned in my [previous comments](https://github.com/MicrosoftDocs/msteams-docs/issues/4730#issuecomment-995772268), we see that "shift+tab" is taking the focus to the latest card from second time onwards. Could you please confirm this behavior once?

**Comment by prajwalmanju** [comment]
Hello Meghana,

Yes, as per your earlier comment "shift+tab" is taking the focus to the latest card from second time onwards but, it should be happen on first time itself otherwise screen reader users might get confuse which card was selected.

Thanks!

**Comment by Meghana-MSFT** [comment]
We have raised aa bug for the same. I will post here if there are any updates on the bug.

**Comment by Meghana-MSFT** [comment]
Engineering team has confirmed that this behavior is by design.

**Comment by Nivedipa-MSFT** [comment]
We are closing this issue for now. If needed, please feel free to reopen it.
